source/Government.cpp

```diff
@@ -316,11 +343,27 @@ double Government::InitialPlayerReputation() const
 // Get the amount that your reputation changes for the given offense.
 double Government::PenaltyFor(int eventType) const
 {
-	double penalty = 0.;
-	for(const auto &it : penaltyFor)
-		if(eventType & it.first)
-			penalty += it.second;
-	return penalty;
+	return PenaltyHelper(eventType, penaltyFor);
+}
+
+
+
+double Government::PenaltyFor(int eventType, const Government *other) const
+{
+	if(!other)
+		return 0.;
+
+	if(other == this)
+		return PenaltyFor(eventType);
+
+	const Government &gov = *other;
```

Comment:
I fail to see the point in getting a ref instead of a pointer
just insta get the id?

Reply:
The government is used in places other than just to get the id.

Reply:
What I meant is we already have said government and just convert it here